lnc/lnc.go

```diff
@@ -0,0 +1,169 @@
+package lnc
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/lightninglabs/lightning-node-connect/mailbox"
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/mwitkow/grpc-proxy/proxy"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+// HeaderMacaroon is the HTTP header field name that is used to send
+// the macaroon.
+const HeaderMacaroon = "Macaroon"
+
+type conn struct {
+	client     lnrpc.LightningClient
+	grpcClient *grpc.ClientConn
+	creds      credentials.PerRPCCredentials
+	cancel     func()
+}
+
+func (c *conn) Close() error {
+	err := c.grpcClient.Close()
+	c.cancel()
+	return err
+}
+
+type LNC struct {
```

Comment:
`lnc.NodeConnection`? 

`lnc.NodeConn`

`lnc.Node`

Reply:
NodeConn it is :+1: